sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -159,13 +179,58 @@ public UrlBuilder setQuery(String query) {
         return this;
     }
 
+    /**
+     * Clear the query that will be used to build the final URL.
+     *
+     * @return This UrlBuilder so that multiple setters can be chained together.
+     */
+    public UrlBuilder clearQuery() {
+        if (query != null && !query.isEmpty()) {
```

Comment:
Is there a case where `query` would ever be null? Also, I think we need to have guards on clearing as it should be idempotent.

Reply:
Solved.